src/main/java/frc/robot/subsystems/swerve/SwerveSubsystem.java

```diff
@@ -211,4 +214,13 @@ public void resetPosition() {
     private Rotation2d getGyroHeading() {
         return Rotation2d.fromDegrees(-ahrs.getAngle());
     }
+
+    private Rotation2d getGyroForField() {
+        return Rotation2d.fromDegrees(-(ahrs.getAngle() - angleoffset));
+    }
+
+    public void resetFieldAngle() {
```

Comment:
the final challenge is abstracting this code (that's common between the two years) too :))))

Reply:
The states, gyro, and `setSwerveDrivePowers` can be abstracted too! Maybe even periodic? The hardest part would be abstracting the modules and adding odometry to 2020.